app/src/org/commcare/CommCareApplication.java

```diff
@@ -1206,4 +1209,14 @@ public boolean isNsdServicesEnabled() {
     public void customiseOkHttp() {
         CommCareNetworkServiceGenerator.customizeRetrofitSetup(new OkHttpBuilderCustomConfig());
     }
+
+    private void setRxJavaGlobalHandler() {
+        RxJavaPlugins.setErrorHandler(throwable -> {
+            // Swallow UndeliverableException
+            if (throwable instanceof UndeliverableException) {
+                return;

```

Comment:
Instead of completely swallowing the error we should deliver the wrapped exception to the uncaught ex handler similar to [here](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling)

Reply:
[There](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling), they don't deliver the exception to the handler, they log it, right?

Reply:
They deliver it to handler - 

````
  if (e instanceof UndeliverableException) {
        e = e.getCause();
  }
  // further checks on new e retrieved from above
````

Reply:
But we should swallow when the wrapped exception is `RequestAbandonedException`

Reply:
whats the reasoning behind swallowing `RequestAbandonedException` ?

Reply:
We throw it when the Thread is in the interrupted state and I didn't see any exception handling around it, so it seems that it was being swallowed in the previous versions of RxJava and that was intentional. Right?

Reply:
I don't think it was intentional, unless there is a reason to swallow it we should do the same handling for it as well.

Reply:
Let me see if I can find any context on that, I also don't see any messages associated with that exception.

Reply:
Confirming that `RequestAbandonedException` is intended to kill its thread (or allow it to terminate its semantic operation and return to a thread pool) as quickly as possible without bubbling back up to the main thread, so swallowing in this fashion is inline with expectation.

Reply:
makes sense, thanks for clarification. Just to clarify It doesn't seem like we were swallowing this earlier and it would have been crashing the app which I agree doesn't sound like right behaviour.

Reply:
I think RxJava was swallowing as part of its behaviour something that changed in later versions as described [here](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling). What puzzles me is why this only started manifesting when we bumped to version `2.2.x`, from a quick search in Crashlytics we can see that crashes only started when we bumped to that version.